rest-service/manager_rest/load_permissions.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import logging
+import os
+
+import yaml
+
+from manager_rest import config
+from manager_rest.flask_utils import setup_flask_app
+from manager_rest.storage.models_base import db
+
+
+def _update_roles(new_roles, keep):
+    existing_roles = {r.name: r for r in
+                      db.session.execute('select id, type, name from roles')}
+    logging.info('Found %d existing roles', len(existing_roles))
+    logging.debug('Found roles: %s', existing_roles)
+    new_roles = {r['name']: r for r in new_roles}
+    roles_to_drop = set(existing_roles) - set(new_roles)
+    roles_to_add = set(new_roles) - set(existing_roles)
+    for role in roles_to_add:
+        db.session.execute(
```

Comment:
Should we maybe turn off autocommit for this so that we get all or nothing, rather than partially inserted data?

Reply:
Oh, or do we do that? I see you have a commit elsewhere, but I thought the db object would automatically make transactions?

Reply:
oh, in flask-sqla, autocommit is a thing tied to requests and to the request context; it has no power here!